projects/node-http-exercise/node-http-exercise.md

```diff
@@ -0,0 +1,255 @@
+
+- To build **an API with a given set of endpoints**.
+- Working with APIs is the daily bread of any full stack developer. Although the Express module is highly used in the industry and makes things much easier, to learn how to build one from scratch will pay you back in spades, and can be really useful to you further down the road.
+
+### Secondary Goals
+
+- You will learn solely rely upon Node HTTP server, without using Express, learning to rely upon knowledge instead of dependencies.
+- You will learn how to setup different routes.
+- You will hone your skills in **Node**.
+
+## Helpful Resources
+
+- The entire [Week 6](https://github.com/Techtonica/curriculum#week-6---web-apis) lessons, except #9 which brings in Express. I would particularly revise #1-2-3-4 as they cover the core of the principles behind this assignment.
+- The [official Node documentation](https://nodejs.org/api/http.html), which will be helpful in so many ways.
+
+## Requirements
+
+- Knowledge of HTTP, APIs, Node.
+- Knowledge of Postman, although the tests are already handed out to you in the `node-http-exercise-postman-collection.json` file and the procedure to import it and run the tests detailed in the [Testing](#Testing) section.
+- Knowledge of ES6 syntax, especially [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions). [Destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) can also help you produce cleaner code.
+- When working with APIs, it is important to use the common status codes to build your response. As a reminder, here are the most common server responses:
+
+| code | status                                    |
+| ---- | ----------------------------------------- |
+| 1XX  | Informational                             |
+| 2XX  | Success                                   |
+| 200  | OK                                        |
+| 201  | OK created                                |
+| 204  | No content, but the headers may be useful |
+| 3XX  | Redirection                               |
+| 301  | Moved to new URL                          |
+| 302  | Found                                     |
+| 304  | Not Modified (Cashed Version)             |
+| 4XX  | Client Error                              |
+| 400  | Bad Request                               |
+| 401  | Unauthorized                              |
+| 404  | Not Found                                 |
+| 5XX  | Server Error                              |
+| 500  | Internal server error                     |
+
+_For a more comprehensive list, please refer to the [official MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status)_
+
+## Instructions
+
+### Step 1 - Create HTTP server
+
+1. Create a `server.js` file in the folder you wish to work in.
+1. In this file, import, using `require()`, the core http module.
+1. In this file, import also the core url module.
```

Comment:
I would just make these part of a provided starter snippet (see below).

Reply:
fixed in [799330c](https://github.com/Techtonica/curriculum/pull/1406/commits/799330cf2b82a0e0d74d3cd2e258f97adc3e70db)

Reply:
Totally forgot this one. I can fix it.

Reply:
OK, fixed in [fef0300](https://github.com/Techtonica/curriculum/pull/1406/commits/fef03008eedd1d995ed0ef7a73e62137ab1b3dda).